WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift

```diff
@@ -403,6 +405,185 @@ final class DashboardViewModelTests: XCTestCase {
             viewModel.hasOrders == true
         }
     }
+
+    // MARK: Dashboard cards
+
+    func test_generated_default_cards_are_as_expected_with_m2_feature_flag_enabled() {
+        // Given
+        let featureFlagService = MockFeatureFlagService(isDynamicDashboardM2Enabled: true)
+
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores, featureFlags: featureFlagService)
+        mockLoadDashboardCards()
+
+        let expectedCards = [DashboardCard(type: .onboarding, availability: .show, enabled: true),
+                             DashboardCard(type: .performance, availability: .unavailable, enabled: false),
+                             DashboardCard(type: .topPerformers, availability: .unavailable, enabled: false),
+                             DashboardCard(type: .blaze, availability: .hide, enabled: false),
+                             DashboardCard(type: .inbox, availability: .show, enabled: false),
+                             DashboardCard(type: .reviews, availability: .show, enabled: false),
+                             DashboardCard(type: .coupons, availability: .show, enabled: false),
+                             DashboardCard(type: .stock, availability: .show, enabled: false)]
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        // Then
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+
+        XCTAssertEqual(viewModel.dashboardCards, expectedCards)
```

Comment:
Nit: this can be simplify by checking the dashboard cards in `waitUntil`. Similar comment for other similar tests.

```suggestion
        // Then
        waitUntil {
            viewModel.dashboardCards == expectedCards
        }

```